inc/global_content/namespace.php

```diff
@@ -0,0 +1,368 @@
+<?php
+/**
+ * Altis Global Content Repository.
+ *
+ * @package altis/core
+ */
+
+namespace Altis\Global_Content;
+
+use Exception;
+use WP_Admin_Bar;
+use WP_CLI;
+use WP_Site;
+
+/**
+ * Setup global media hooks.
+ *
+ * @return void
+ */
+function bootstrap() : void {
+	// Create the site on upgrade.
+	add_action( 'altis.migrate', __NAMESPACE__ . '\\maybe_create_site' );
+
+	// Bootstrap media site if it exists.
+	add_action( 'muplugins_loaded', __NAMESPACE__ . '\\bootstrap_site', 1 );
+}
+
+/**
+ * Add global content site hooks.
+ *
+ * @return void
+ */
+function bootstrap_site() : void {
+	if ( defined( 'WP_INITIAL_INSTALL' ) && WP_INITIAL_INSTALL ) {
+		return;
+	}
+
+	if ( empty( get_site_id() ) ) {
+		return;
+	}
+
+	// Redirect global site URLs.
+	add_action( 'admin_init', __NAMESPACE__ . '\\redirect_admin_pages' );
+	add_action( 'template_redirect', __NAMESPACE__ . '\\redirect_frontend' );
+
+	// Handle clean up operations.
+	add_action( 'wp_uninitialize_site', __NAMESPACE__ . '\\uninitialize_media_site' );
+
+	// Handle global content repo admin customisations.
+	add_action( 'admin_menu', __NAMESPACE__ . '\\admin_menu', 1000 );
+	add_action( 'admin_bar_menu', __NAMESPACE__ . '\\admin_bar_menu', 1000 );
+
+	// Handle global site URL changes.
+	add_action( 'updated_option_siteurl', __NAMESPACE__ . '\\handle_siteurl_update', 10, 2 );
+	add_action( 'updated_option_home', __NAMESPACE__ . '\\handle_siteurl_update', 10, 2 );
+	add_action( 'wp_update_site', __NAMESPACE__ . '\\handle_site_update' );
+
+	// Do not allow global site deletion.
+	add_filter( 'map_meta_cap', __NAMESPACE__ . '\\prevent_site_deletion', 10, 4 );
+
+	// Handle network admin sites list.
+	add_filter( 'manage_sites_action_links', __NAMESPACE__ . '\\sites_list_row_actions', 10, 2 );
+	add_filter( 'display_site_states', __NAMESPACE__ . '\\add_global_site_state', 10, 2 );
+}
+
+/**
+ * Get the global content repository site ID.
+ *
+ * @return integer|null
+ */
+function get_site_id() : ?int {
+	return get_site_option( 'global_content_site_id', null );
```

Comment:
This fails to identify if a global content site was initialized but then manually changed to a static url in the config. It also fails to identify manually-set Global Content Site URLs just generally, so a `wp altis migrate` after manually setting the URL to the Global Content Site in the config will create a new `/repo/` site but no site is set as the "global_content_site".

Reply:
If it was manually changed the WP AMF source will use that manual setting and it will take precedence. This global content site might still be used by other modules for something like global reusable blocks for example.